ThunderCloud/DeveloperModeController.swift

```diff
@@ -84,7 +84,9 @@ public class DeveloperModeController: NSObject {
         
         backgroundObserver = NotificationCenter.default.addObserver(forName: UIApplication.willEnterForegroundNotification, object: nil, queue: OperationQueue.main, using: { [weak self] (notification) in
             
-            if DeveloperModeController.devModeOn {
+            // Make sure we have an app I and app.json before entering developer mode, now that we have apps who download their bundle rather
+            // than being bundled with it.
+            if DeveloperModeController.devModeOn, ContentController.shared.fileExistsInBundle(file: "app.json"),  UserDefaults.standard.string(forKey: "TSCAppId") ?? API_APPID != nil {
```

Comment:
Wondering if `ContentController.shared.fileExistsInBundle(file: "app.json"),  UserDefaults.standard.string(forKey: "TSCAppId") ?? API_APPID != nil` would be better sat in a guard above this, with the `checkForUpdates` section on lines 95-97 being deferred? That way it'd be more explicit that before logging into dev mode / switching to live we must have those items.

Also, are we able to check for updates without those items? If so, a guard (and no defer) could be good? :)

Reply:
@ryanbourneuk that's a very good point... Have guarded now against all the code